server/index.go

```diff
@@ -140,40 +160,96 @@ func (item *Item) Category() string {
 
 // SizeString returns a human-readable string representing the size of the in iRODS.
 func (item *Item) SizeString() string {
-	if item.Size < 1024 {
-		return fmt.Sprintf("%d B", item.Size)
+	size := float64(item.Size)
+	kib := float64(1024)
+	mib := 1024 * kib
+	gib := 1024 * mib
+
+	switch {
+	case size < kib:
+		return fmt.Sprintf("%.0f B", size)
+	case size < mib:
+		return fmt.Sprintf("%.2f KiB", size/kib)
+	case size < gib:
+		return fmt.Sprintf("%.2f MiB", size/mib)
+	default:
+		return fmt.Sprintf("%.2f GiB", size/gib)
 	}
-
-	return fmt.Sprintf("%d KiB", item.Size/1024)
 }
 
 // MetadataStrings returns a sorted list of strings representing the metadata of the item.
-func (item *Item) MetadataStrings() []string {
-	var meta []string
-	for _, m := range item.Metadata {
-		meta = append(meta, fmt.Sprintf("%s=%s", m.Name, m.Value))
+// The metadata are filtered by the given functions, which are applied to each metadata
+// item. If a function returns true, the metadata item is excluded from the result.
+func (item *Item) MetadataStrings(filter ...func(types.IRODSMeta) bool) []string {
+	meta := make([]string, 0, len(item.Metadata))
+
+	for _, avu := range item.Metadata {
+		excluded := false
+		for _, f := range filter {
+			if f(*avu) {
+				excluded = true
+				break
+			}
+		}
+		if !excluded {
+			meta = append(meta, fmt.Sprintf("%s=%s", avu.Name, avu.Value))
+		}
 	}
 	slices.Sort(meta)
 
 	return meta
 }
 
-// ACLStrings returns a sorted list of strings representing the ACL of the item.
-func (item *Item) ACLStrings() []string {
-	var acl []string
-	for _, a := range item.ACL {
-		acl = append(acl, fmt.Sprintf("%s#%s:%s", a.UserName, a.UserZone, a.AccessLevel))
+// ACLStrings returns a sorted list of strings representing the ACL of the item. The ACL
+// is filtered by the given functions, which are applied to each access item. If a
```

Comment:
```suggestion
// is filtered by the given functions, which are applied to each access item. If a 
// function returns true, the access item is excluded from the result.
```
missing half a sentence here